### PR TITLE
Add `os_family`, `os_platform` and `os_version` to host provider

### PR DESCRIPTION
**Pull request by belimawr**
## What does this PR do?
This commit adds `os_family`, `os_platform` and `os_version` to the host provider, enabling differentiating Linux distributions. This is required to support Debian 12 and other distributions that are moving away from traditional log files in favour of Journald.

## Why is it important?
It is required to support Debian 12 and other Linux distributions that are moving away from traditional log files in favour of Journald.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Disruptive User Impact

None

## How to test this PR locally

1. Package the Elastic-Agent
2. Deploy it with any policy, even an empty policy with no monitoring will work
3. Collect the diagnostics
4. Look at `variables.yml` and ensure `os_family`, `os_platform` and `os_version` are populated under `host`. Here is an example from Debian 11:
```
      host:
        architecture: x86_64
        id: 9a7aa218396d47b1b17218a99918bae4
        ip:
            - 127.0.0.1/8
            - 10.0.2.15/24
            - ::1/128
            - fe80::a00:27ff:fe8d:c04d/64
        mac:
            - 08:00:27:8d:c0:4d
        name: bullseye
        os_family: debian
        os_platform: debian
        os_version: 11 (bullseye)
        platform: linux
```

## Related issues

- Relates https://github.com/elastic/integrations/issues/10797
- Is required by https://github.com/elastic/integrations/pull/11618

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Comment by pkoutsovasilis** [comment]
@belimawr could you add a changelog fragment? 🙂

**Review by swiatekm — APPROVED** [review]

File: internal/pkg/composable/providers/host/host_test.go [hunk 1]
```diff
@@ -167,3 +168,20 @@ func returnHostMapping(log *logger.Logger) infoFetcher {
 		return host, nil
 	}
 }
+
+func TestGetHostInfoReturnsSomeKeys(t *testing.T) {
+	// this is a simple test to ensure the host provicer returns the new keys
```
Comment (mauri870) [inline internal/pkg/composable/providers/host/host_test.go:173]:
```suggestion
	// this is a simple test to ensure the host provider returns the new keys
```

File: internal/pkg/composable/providers/host/host_test.go [hunk 2]
```diff
@@ -167,3 +168,20 @@ func returnHostMapping(log *logger.Logger) infoFetcher {
...
+		t.Fatal(err)
+	}
+
+	expectedKeys := []string{"os_family", "os_platform", "os_version"}
+
+	for _, key := range expectedKeys {
+		if _, exist := osInfo[key]; !exist {
+			t.Errorf("expecting key '%s' from host provider.", key)
+		}
+	}
```
Comment (mauri870) [inline internal/pkg/composable/providers/host/host_test.go:186]: I'm not sure if this is needed, but should we also check that the values are not empty?

Comment (belimawr) [reply internal/pkg/composable/providers/host/host_test.go:186]: That's highly dependant on the OS/Linux distro it's running on, for the few I've tested, only Arch had an empty string, but overall it's very inconsistent across different distros (some examples: https://github.com/elastic/integrations/issues/10797#issuecomment-2452212459).

File: internal/pkg/composable/providers/host/host_test.go [hunk 3]
```diff
@@ -167,3 +168,20 @@ func returnHostMapping(log *logger.Logger) infoFetcher {
 		return host, nil
 	}
 }
+
+func TestGetHostInfoReturnsSomeKeys(t *testing.T) {
+	// this is a simple test to ensure the host provider returns the new keys
+	// needed to add support to Debian 12.
+	osInfo, err := getHostInfo(logp.L())()
```
Comment (AndersonQ) [inline internal/pkg/composable/providers/host/host_test.go:175]: There is a test logger on https://github.com/elastic/elastic-agent/blob/d8125be16f8bf369738113861f958c59441bc696/pkg/core/logger/loggertest/logger.go#L20. Better to use it. Or at least the logger from `pkg/core/logger/`

**Comment by belimawr** [comment]
> @belimawr could you add a changelog fragment? 🙂

Done on 2a789ba764

**Review by mauri870 — APPROVED** [review]

**Comment by strawgate** [comment]
Per ECS -- https://www.elastic.co/guide/en/ecs/current/ecs-os.html these should be `os.family`, `os.platform`, `os.version`?

**Comment by strawgate** [comment]
Oh, if these don't end up in metadata for documents then you're right it doesnt need to be ecs

**Pull request merged**